diffpy/snmf/subroutines.py

```diff
@@ -84,6 +84,33 @@ def construct_stretching_matrix(components, number_of_components, number_of_sign
     return stretching_factor_matrix
 
 
+def construct_component_matrix(components, signal_length):
+    """Constructs the component matrix
+
+    Parameters
+    ----------
+    components: tuple of ComponentSignal objects
+      The tuple containing the component signals in ComponentSignal objects.
+    signal_length: int
+      The number of signals in the data provided by the user.
+
+    Returns
+    -------
+    2d array
+      The matrix containing the component signal values. Has dimensions `signal_length` x `number_of_components`.
+
+    """
+    if signal_length <= 0:
```

Comment:
isn't `signal_length = len(components[0].iq)`?

So you don't need to pass it in either.

Reply:
That's true. I will change it